### PR TITLE
Support remark-lint installed locally

### PR DESCRIPTION
**Pull request by watagashi**
Support remark-lint installed locally and enables lint without saving it to disk.
It is like textlint.

### CODE REVIEW

File: autoload/ale/handlers/remark_lint.vim [hunk 1]
```diff
@@ -0,0 +1,43 @@
+" Description: remark-lint for Markdown files
```
Comment (w0rp) [inline autoload/ale/handlers/remark_lint.vim]: Could you move these functions back to where they were before? These changes aren't needed.

File: test/command_callback/test_remark_lint_command_callbacks.vader [hunk 2]
```diff
@@ -0,0 +1,64 @@
+Before:
```
Comment (w0rp) [inline test/command_callback/test_remark_lint_command_callbacks.vader]: Rebase your commits, and look at how linter tests are written now. If you move the functions back to where they were, you should be able to make this file a lot smaller. See `:help ale-development-linter-tests` in the latest version of ALE.

File: autoload/ale/handlers/remark_lint.vim [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+"
+call ale#Set('remark_lint_executable', 'remark')
+call ale#Set('remark_lint_use_global', get(g:, 'ale_use_global_executables', 0))
+call ale#Set('remark_lint_options', '')
```
Comment (w0rp) [inline autoload/ale/handlers/remark_lint.vim]: Don't forget to document these new options.

**Comment by watagashi** [comment]
@w0rp Thanks your feedback.

I opened new pull request #1730 based on this PR and feedback, because changes on master is huge and rebasing this PR is complex.

I'll close this request.

**Pull request closed without merging**